error compiling core 1.142

when trying to upgrade core to 1.142.1, we get the following build error.

1.142.0 is failing as well, 1.141.2 is fine.

used toolchain picked up from [`./rust-toolchain`](https://github.com/deltachat/deltachat-ios/blob/main/rust-toolchain) is 1.77.0 - not sure if that is related, according to [CHANGELOG](https://github.com/deltachat/deltachat-core-rust/blob/main/CHANGELOG.md), ci is using 1.80.0 since 1.142.1, i will try that as well. EDIT: did so without success, see below for details

EDIT: ftr, android builds 1.142.1 with rust-toolchain 1.77.0

cc @link2xt 

```
Showing All Issues
error[E0061]: this function takes 2 arguments but 1 argument was supplied

   --> src/push.rs:65:24

    |

65  |         let response = http::get_client(socks5_config)?

    |                        ^^^^^^^^^^^^^^^^--------------- an argument of type `bool` is missing

    |

/Users/bpetersen/projects/deltachat-ios/deltachat-ios/libraries/deltachat-core-rust/Cargo.toml: unused manifest key: lints.rust.unexpected_cfgs.check-cfg

   --> src/push.rs:65:41

    |

65  |         let response = http::get_client(socks5_config)?

    |                                         ^^^^^^^^^^^^^

    = note: expected reference `&context::Context`

                    found enum `std::option::Option<_>`

expected `&Context`, found `Option<_>`

   --> src/net/http.rs:135:21

    |

135 | pub(crate) async fn get_client(context: &Context, load_cache: bool) -> Result<reqwest::Client> {

    |                     ^^^^^^^^^^ -----------------  ----------------

help: consider using `Option::expect` to unwrap the `std::option::Option<_>` value, panicking if the value is an `Option::None`

    |

65  |         let response = http::get_client(socks5_config.expect("REASON"))?

    |                                                      +++++++++++++++++

help: provide the argument

    |

65  |         let response = http::get_client(/* &context::Context */, /* bool */)?

    |                                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~



error[E0277]: the `?` operator can only be applied to values that implement `Try`

  --> src/push.rs:65:24

   |

65 |         let response = http::get_client(socks5_config)?

   |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the `?` operator cannot be applied to type `impl futures::Future<Output = std::result::Result<reqwest::Client, anyhow::Error>>`

   |

   = help: the trait `Try` is not implemented for `impl futures::Future<Output = std::result::Result<reqwest::Client, anyhow::Error>>`

help: consider `await`ing on the `Future`

   |

65 |         let response = http::get_client(socks5_config).await?

   |                                                       ++++++



Some errors have detailed explanations: E0061, E0277.

For more information about an error, try `rustc --explain E0061`.

function defined here

[ERROR cargo_lipo] Failed to build "deltachat_ffi" for "x86_64-apple-ios": Executing "/Users/bpetersen/.rustup/toolchains/1.77.0-aarch64-apple-darwin/bin/cargo" "--color" "auto" "build" "--manifest-path" "../deltachat-ios/libraries/deltachat-core-rust/deltachat-ffi/Cargo.toml" "-p" "deltachat_ffi" "--target" "x86_64-apple-ios" "--lib" "--features" "jsonrpc" finished with error status: exit status: 101

could not compile `deltachat` (lib) due to 2 previous errors
```

--- a/src/imap.rs
+++ b/src/imap.rs
@@ -1489,7 +1489,7 @@ impl Session {
         } else if !context.push_subscriber.heartbeat_subscribed().await {
             let context = context.clone();
             // Subscribe for heartbeat notifications.
-            tokio::spawn(async move { context.push_subscriber.subscribe().await });
+            tokio::spawn(async move { context.push_subscriber.subscribe(&context).await });
         }
 
         Ok(())

--- a/src/push.rs
+++ b/src/push.rs
@@ -48,7 +48,7 @@ impl PushSubscriber {
 
     /// Subscribes for heartbeat notifications with previously set device token.
     #[cfg(target_os = "ios")]
-    pub(crate) async fn subscribe(&self) -> Result<()> {
+    pub(crate) async fn subscribe(&self, context: &Context) -> Result<()> {
         use crate::net::http;
 
         let mut state = self.inner.write().await;
@@ -61,8 +61,9 @@ impl PushSubscriber {
             return Ok(());
         };
 
-        let socks5_config = None;
-        let response = http::get_client(socks5_config)?
+        let load_cache = true;
+        let response = http::get_client(context, load_cache)
+            .await?
             .post("https://notifications.delta.chat/register")
             .body(format!("{{\"token\":\"{token}\"}}"))
             .send()
@@ -77,7 +78,7 @@ impl PushSubscriber {
 
     /// Placeholder to skip subscribing to heartbeat notifications outside iOS.
     #[cfg(not(target_os = "ios"))]
-    pub(crate) async fn subscribe(&self) -> Result<()> {
+    pub(crate) async fn subscribe(&self, _context: &Context) -> Result<()> {
         let mut state = self.inner.write().await;
         state.heartbeat_subscribed = true;
         Ok(())

